contracts/smartwallet/CustomSmartWallet.sol

```diff
@@ -14,7 +14,11 @@ contract CustomSmartWallet is IForwarder {
 
     uint256 public override nonce;
     bytes32 public constant DATA_VERSION_HASH = keccak256("2");
-    bytes32 public domainSeparator;    
+    bytes32 public domainSeparator;
+
+    constructor() public {
+        setOwner(msg.sender);
```

Comment:
Wouldn't caling setOwner() here make it impossible to run initialize() afretwards?

Reply:
By initializing it, we are initializing the logic contract and not the proxy which in our case are the smart wallet address created with the index. 

Im still doing to testing to make sure that its working properly but so far so good. I just need to confirm if we need to setup to msg.sender or a different address or if we add an initialization flag, but that increase the gas cost.